### ISSUE TITLE
Can we inform services.items.url with value returned by api endpoint?

### ISSUE THREAD

**Issue by MatthewCroughan** [bastienwirtz/homer#98]
It is possible to set custom messages with api endpoints, but what about the `url:` field for a service? For example, one could imagine a use case where we poll the docker socket for the hostname of the service as we've set it in docker-compose.yml, that way we can change service names and this will always be a relative pointer to `service.local`, for example.

This is just one use case, but what I really want is for it to refer to the IP address of the machine that the webserver serving homer is running on, which seems like it would be impossible if that webserver was running in a container, as it would report the private docker network.

**Comment by bastienwirtz** [comment]
Hey @MatthewCroughan !

:thinking: Could you use relative URL instead of having to change the base url ?

**Comment by MatthewCroughan** [comment]
@bastienwirtz How does one do that? If that's already possible to do, maybe we should write some docs on how to do it!

**Comment by bastienwirtz** [comment]
If you put something like: 
```yaml
- name: "Hello"
  [...]
  url: "/hello/"
```

It's going to behave like a classic relative link (the `url` is just copied in the `href` attribute of the `a` html tag)

**Comment by MatthewCroughan** [comment]
I did not realise this basic fact hah. This solves 99% of my issue, practically. 

However, I still think it'd be neat to have the hostname for the services present, since I plan on running Traefik, a reverse proxy and load balancer you may be familiar with to run all of my services. It's possible to look at the docker socket to return the hostnames of services, and more. It'd perhaps be nice if there could be some sort of interaction there in order to populate these urls automatically based on the running services in your docker-compose.

**Comment by bastienwirtz** [comment]
I see, I'll think about it @MatthewCroughan. I might use homer on the same type of setup in the future. I'm glad the relative URL solution works for most of your issue. I also plan to add different different service "types" which will be able to load some specific logic, that could be usefull for your issue.

**Comment by simonporte** [comment]
> If you put something like: 
> ```yaml
> - name: "Hello"
>   [...]
>   url: "/hello/"
> ```
> 
> It's going to behave like a classic relative link (the `url` is just copied in the `href` attribute of the `a` html tag)

Any way to make this work with ports ?
Relative path assume the port is the same as Homer, but most of my links use custom ports.
It does annoy me because I use hostnames in config and Android does not resolve local hostnames contrary to desktop. So links does not work on phone if they dont use IPs instead of hostnames.
I don't use traefik as I don't need to overcomplicate things but some kind of interaction would be nice.

**Comment by MatthewCroughan** [comment]
@simonporte I do something like this with yml in my project Octobalena. Either script can act as an entrypoint in your docker container. You can then modify the yaml of the homer config to match your desires, which can be informed from outside variables such as the hostnames of your things, as set by a docker .env file. You could even query an API and store the result, then use that result to modify the yml object. 

TL;DR you can make `services.items.url` == whatever you want, with a custom entrypoint script similar to the ones below.

https://github.com/MatthewCroughan/octobalena/blob/master/octodash/setapikey.py#L1-L11
```python3
import json
import os

with open('/root/.config/octodash/config.json', 'r+') as f:
    json_data = json.load(f)
    json_data["config"]["octoprint"]["accessToken"] = os.environ["OCTOPRINT_APIKEY"]
    f.seek(0)
    f.write(json.dumps(json_data, indent=4))
    f.truncate()
```
https://github.com/MatthewCroughan/octobalena/blob/master/start.sh#L24-L28
```sh
if [[ -z "$PRINTER_NAME" ]]; then
  jq ".config.printer.name |= \"Octobalena\"" < $FILE > $TMP && mv $TMP $FILE
else
  jq ".config.printer.name |= \"$PRINTER_NAME\"" < $FILE > $TMP && mv $TMP $FILE
fi
```

**Comment by bastienwirtz** [comment]
Hi there. 
It's now possible to implement custom service card that can help with that. See the updated configuration example with Pi-Hole. If you add the "type: PiHole" to the configuration, the PiHole component, with some custom logic (displays pihole status in the dashboard) will be loaded. 
So you could add a component that fetch the information you need.
Hope it can help!
